addon/components/o-s-s/power-select.ts

```diff
@@ -38,19 +50,47 @@ export default class OSSPowerSelect extends Component<OSSPowerSelectArgs> {
   }
 
   @action
-  toggleSelect(event: MouseEvent): void {
-    event.stopPropagation();
-    this.displaySelect = !this.displaySelect;
+  handleSelectorClose(event: ToggleEvent & { target: HTMLDetailsElement }) {
+    if (!event.target.open && document.querySelector(`#${this.portalId}`)) {
+      document.querySelector(`#${this.portalId}`)!.remove();
+      this.cleanupDrodpownAutoplacement?.();
+      this.closeDropdown();
+    }
+  }
 
-    if (!this.displaySelect) {
+  @action
+  toggleDropdown(event: MouseEvent): void {
+    super.toggleDropdown(event);
+
+    if (!this.isOpen) {
       this.args.onSearch?.('');
+      return;
     }
+
+    scheduleOnce('afterRender', this, () => {
+      const referenceTarget = this.container.querySelector('.upf-power-select__array-container');
+      const floatingTarget = document.querySelector(`#${this.portalId}`);
+
+      if (referenceTarget && floatingTarget) {
+        this.cleanupDrodpownAutoplacement = attachDropdown(
+          referenceTarget as HTMLElement,
+          floatingTarget as HTMLElement
+        );
+      }
+    });
   }
 
   @action
-  hideSelect(_: HTMLElement, event: MouseEvent): void {
-    event.stopPropagation();
-    this.displaySelect = false;
+  onClickOutside(_: HTMLElement, event: MouseEvent): void {
+    super.onClickOutside(_, event);
+    this.cleanupDrodpownAutoplacement?.();
     this.args.onSearch?.('');
+    document.querySelector(`#${this.portalId}`)?.remove();
+  }
+
+  @action
+  registerContainer(element: HTMLDetailsElement): void {
+    super.registerContainer(element);
+    this.portalTarget = isTesting() ? this.container : document.body;
```

Comment:
why not leave it in the body even when testing ? 🤔

Reply:
messes up with the assert.dom method ?
I'm sure we can still document.queryElementBy* though 🤷